modules/sdk-coin-xrp/src/ripple.ts

```diff
@@ -56,7 +53,11 @@ const signWithPrivateKey = function (txHex, privateKey, options) {
       SigningPubKey: publicKey,
       TxnSignature: computeSignature(tx, privateKey, options.signAs),
     };
-    tx.Signers = [{ Signer: signer }];
+    if (tx.TxnSignature || tx.Signers) {
+      tx.Signers.unshift({ Signer: signer });
+    } else {
+      tx.Signers = [{ Signer: signer }];
+    }
```

Comment:
Some comments on ordering between sub-seqeuent function calls would be nice. (when do we expect the if case, when do we expect the else case)